main.nf

```diff
@@ -198,72 +249,328 @@ process get_software_versions {
+     file fasta_decoy from fastafile_decoy
+
+    output:
+     set mzmlID, file("${mzmlID}.idXML") into id_files
+
+    script:
+     """
+     ${openms_path}/CometAdapter -comet_executable ${comet_executable_path} -in ${mzml_file} -out ${mzmlID}.idXML -threads ${params.num_threads} -database ${fasta_decoy} -precursor_mass_tolerance ${params.pmt} -fragment_bin_tolerance ${params.fmt} -fragment_bin_offset ${params.fbo} -num_hits ${params.num_hits} -digest_mass_range ${params.dmr} -max_variable_mods_in_peptide ${params.maxmod} -allowed_missed_cleavages 0 -precursor_charge ${params.prec_charge} -activation_method ${params.activ_method} -use_NL_ions true -variable_modifications '${params.variable_mods}' -fixed_modifications ${params.fixed_mods} -enzyme '${params.enzyme}'
+     """
 
+}
+
+
+/*
+ * STEP 3 - index decoy and target hits
+ */
+process index_peptides {
+    publishDir "${results_path}/"
+ 
     input:
-    set val(name), file(reads) from read_files_fastqc
+     set ID_idx, file(id_file) from id_files
+     file fasta_decoy from fastafile_decoy
 
     output:
-    file "*_fastqc.{zip,html}" into fastqc_results
+     set ID_idx, file("${ID_idx}_idx.idXML") into id_files_idx, id_files_idx_original
 
     script:
-    """
-    fastqc -q $reads
-    """
+     """
+     ${openms_path}/PeptideIndexer -in ${id_file} -out ${ID_idx}_idx.idXML -threads ${params.num_threads} -fasta ${fasta_decoy} -decoy_string DECOY -enzyme:specificity none
+     """
+
 }
 
 
+/*
+ * STEP 4 - calculate fdr for id based alignment
+ */
+process calculate_fdr_for_idalignment {
+    publishDir "${results_path}/"
+ 
+    input:
+     set ID_idx_fdr, file(id_file_idx) from id_files_idx
+
+    output:
+     set ID_idx_fdr, file("${ID_idx_fdr}_fdr.idXML") into id_files_idx_fdr
+
+    script:
+     """
+     ${openms_path}/FalseDiscoveryRate -in ${id_file_idx} -out ${ID_idx_fdr}_fdr.idXML -threads ${params.num_threads}
+     """
+
+}
+
 
 /*
- * STEP 2 - MultiQC
+ * STEP 5 - filter fdr for id based alignment
  */
-process multiqc {
-    publishDir "${params.outdir}/MultiQC", mode: 'copy'
+process filter_fdr_for_idalignment {
+    publishDir "${results_path}/"
+ 
+    input:
+     set ID_idx_fdr_filtered, file(id_file_idx_fdr) from id_files_idx_fdr
+
+    output:
+     set ID_idx_fdr_filtered, file("${ID_idx_fdr_filtered}_filtered.idXML") into id_files_idx_fdr_filtered
+
+    script:
+     """
+     ${openms_path}/IDFilter -in ${id_file_idx_fdr} -out ${ID_idx_fdr_filtered}_filtered.idXML -threads ${params.num_threads} -score:pep 0.05  -remove_decoys
+     """
+
+}
+
+
+/*
+ * STEP 6 - compute alignment rt transformation
+ */
+process align_ids {
+   publishDir "${results_path}/"
 
     input:
-    file multiqc_config
-    file ('fastqc/*') from fastqc_results.collect()
-    file ('software_versions/*') from software_versions_yaml
-    file workflow_summary from create_workflow_summary(summary)
+     file id_names from id_files_idx_fdr_filtered.collect{it[1]}
```

Comment:
That requires some tuning of the channel - accessing files that way will break at some point when you have multiple files as the sorting is not always identical. E.g. nextflow doesn't take care which file will be outputted first on the id_files_idx_fdr_filtered channel for example, which might cause unexpected side effects in general!